src/route-handlers/describe-workflow/describe-workflow.ts

```diff
@@ -23,21 +28,119 @@ export default async function describeWorkflow(
         runId: decodedParams.runId,
       },
     });
+    const res: DescribeUnArchivedWorkflowResponse = merge(
+      {},
+      describeResponse,
+      {
+        workflowExecutionInfo: { closeEvent: null, isArchived: false as const },
+      }
+    );
+    if (
+      res.workflowExecutionInfo &&
+      res.workflowExecutionInfo.closeStatus &&
+      res.workflowExecutionInfo.closeStatus !==
+        'WORKFLOW_EXECUTION_CLOSE_STATUS_INVALID'
+    ) {
+      const closeEventResponse = await grpcClient.clusterMethods[
+        decodedParams.cluster
+      ].getHistory({
+        domain: decodedParams.domain,
+        workflowExecution: {
+          workflowId: decodedParams.workflowId,
+          runId: decodedParams.runId,
+        },
+        historyEventFilterType: 'EVENT_FILTER_TYPE_CLOSE_EVENT',
+      });
+      if (closeEventResponse.history?.events?.[0])
+        res.workflowExecutionInfo.closeEvent =
+          closeEventResponse.history?.events?.[0];
+    }
 
     return NextResponse.json(res);
   } catch (e) {
-    logger.error<RouteHandlerErrorPayload>(
-      { requestParams: decodedParams, cause: e },
-      'Error fetching workflow info'
-    );
+    //describe workflow depends on a temp datasource, so sometimes data is not available
+    // to make it more reliable we depend on history to construct similar response in case data is not available
+    try {
+      if (e instanceof GRPCError && e.httpStatusCode !== 404) {
+        throw e;
+      }
+      const archivedHistoryResponse = await grpcClient.clusterMethods[
+        decodedParams.cluster
+      ].getHistory({
+        domain: decodedParams.domain,
+        workflowExecution: {
+          workflowId: decodedParams.workflowId,
+          runId: decodedParams.runId,
+        },
+        pageSize: 1,
+      });
+      const archivedHistoryEvents =
+        archivedHistoryResponse.history?.events || [];
 
-    return NextResponse.json(
-      {
-        message:
-          e instanceof GRPCError ? e.message : 'Error fetching workflow info',
-        cause: e,
-      },
-      { status: getHTTPStatusCode(e) }
-    );
+      if (!archivedHistoryEvents[0]?.workflowExecutionStartedEventAttributes) {
+        throw e;
+      }
+      const { runId, workflowId } = decodedParams;
```

Comment:
nit: destructuring decodedParams is unnecessary imo since we don't use runId and workflowId in multiple places after this. If you want to destructure it, you can do it in the beginning of the route handler itself.